llvm/include/llvm/SandboxIR/PassManager.h

```diff
@@ -65,6 +66,34 @@ class FunctionPassManager final
   bool runOnFunction(Function &F) final;
 };
 
+/// Owns the passes and provides an API to get a pass by its name.
+class PassRegistry {
+  SmallVector<std::unique_ptr<Pass>, 8> Passes;
+  DenseMap<StringRef, Pass *> NameToPassMap;
+
+public:
+  PassRegistry() = default;
+  /// Registers \p PassPtr and takes ownership.
+  Pass *registerPass(std::unique_ptr<Pass> &&PassPtr) {
```

Comment:
maybe this should return a reference instead of a pointer since it'll never be null?

Reply:
Yes a reference makes more sense.